### ISSUE TITLE
Path `./jsconfig.paths.json` does not exist preventing nodejs container from starting

### ISSUE THREAD

**Issue by DistantVoyager** [nautobot/nautobot#4124]
### Environment
* Nautobot version: 2.0.0 (next-[8413399](https://github.com/nautobot/nautobot/commit/84133994d6e2bd7566b71383142f0ae4e9458daf))
* Python version: 3.10.6
* Database platform, version: N/A
* Middleware(s): N/A

### Steps to Reproduce
1. Clone the nautobot `next` branch (currently at commit [8413399](https://github.com/nautobot/nautobot/commit/84133994d6e2bd7566b71383142f0ae4e9458daf))
2. `invoke build debug`

### Expected Behavior
Nautobot development server / nodejs docker container starts successfully and can be reached at localhost:3000.

### Observed Behavior
The Nautobot server on localhost:8080 starts successfully; however, the dev container on port 3000 does not. When trying to force start the nautobot-nodejs-1 container, this error is given: 
`react-app-alias:readConfig: '${confPath}' field 'extends' is '${extPath}' - not exists'`

Full debug log:
```
> nautobot@2.0.0-beta.2 start
nautobot-nodejs-1         | > craco start
nautobot-nodejs-1         | 
nautobot-nodejs-1         | /opt/nautobot/ui/node_modules/react-app-alias/src/index.js:147
nautobot-nodejs-1         |     throw Error(`react-app-alias:readConfig: '${confPath}' field 'extends' is '${extPath}' - not exists'`)
nautobot-nodejs-1         |           ^
nautobot-nodejs-1         | 
nautobot-nodejs-1         | Error: react-app-alias:readConfig: '/opt/nautobot/ui/jsconfig.json' field 'extends' is '/opt/nautobot/ui/jsconfig.paths.json' - not exists'
nautobot-nodejs-1         |     at readConfig (/opt/nautobot/ui/node_modules/react-app-alias/src/index.js:147:11)
nautobot-nodejs-1         |     at defaultOptions (/opt/nautobot/ui/node_modules/react-app-alias/src/index.js:189:16)
nautobot-nodejs-1         |     at aliasWebpack (/opt/nautobot/ui/node_modules/react-app-alias-ex/src/index.js:132:20)
nautobot-nodejs-1         |     at Object.overrideWebpackConfig (/opt/nautobot/ui/lib/nautobot-craco-alias-plugin.js:31:16)
nautobot-nodejs-1         |     at overrideWebpack (/opt/nautobot/ui/node_modules/@craco/craco/dist/lib/features/plugins.js:36:38)
nautobot-nodejs-1         |     at /opt/nautobot/ui/node_modules/@craco/craco/dist/lib/features/plugins.js:53:29
nautobot-nodejs-1         |     at Array.forEach (<anonymous>)
nautobot-nodejs-1         |     at applyWebpackConfigPlugins (/opt/nautobot/ui/node_modules/@craco/craco/dist/lib/features/plugins.js:52:29)
nautobot-nodejs-1         |     at mergeWebpackConfig (/opt/nautobot/ui/node_modules/@craco/craco/dist/lib/features/webpack/merge-webpack-config.js:119:70)
nautobot-nodejs-1         |     at overrideWebpackDev (/opt/nautobot/ui/node_modules/@craco/craco/dist/lib/features/webpack/override.js:8:80)
nautobot-nodejs-1         | 
nautobot-nodejs-1         | Node.js v18.16.1
nautobot-nodejs-1 exited with code 1
```

This path `./jsconfig.paths.json` within `/nautobot/ui/jsconfig.json` does not exist and should probably be something else. Full `jsconfig.json` file:
```
{
    "extends": "./jsconfig.paths.json",
    "compilerOptions": {},
    "exclude": ["node_modules", "build"]
}
```

**Comment by glennmatthews** [comment]
Darn! I could have sworn I fixed this in #4082. `jsconfig.paths.json` is supposed to be generated by the `nautobot-server build_ui` command that is run as part of the Docker build.

**Comment by glennmatthews** [comment]
I missed this in #4082. The issue is that while the files are in `/opt/nautobot/ui/` in the base image, `docker-compose.dev.yml` mounts the local source code on top of this directory, clobbering those files. Additionally, since the image is running in `/opt/nautobot/ui/` rather than `/source/nautobot/ui/`, it won't automatically pick up changes to those files and do live rebuilds. Mea culpa.

This is going to be tricky to fix since we can mount volumes for directories (e.g., preserving `/opt/nautobot/ui/node_modules/` within the image when we mount local `nautobot/ui/` onto `/opt/nautobot/ui/`, but not for files - we need some way to "merge" the container's `/opt/nautobot/ui/jsconfig.paths.json` into the volume mounted from localhost, ideally without having that file get exported back to the local filesystem since it should never be committed to git. Similar problem exists for the other two generated files `nautobot/ui/src/app_config.js` and `nautobot/ui/src/app_routes.json`.

**Comment by glennmatthews** [comment]
Should be resolved by #4130.
